### PR TITLE
Ui/player joined screen

### PR DESCRIPTION
**Pull request by m4rtint**
For the most part it works. 

Could do with a lot of clean up. But it works.

Demo:
https://imgur.com/a/Njk6oKK

### CODE REVIEW

**Comment by m4rtint** [comment]
[UI Changes to Player Selection Screen](https://trello.com/c/R3UyENfB/224-ui-changes-to-player-selection-screen)

**Comment by m4rtint** [comment]
[UI Changes to Player Selection Screen](https://trello.com/c/R3UyENfB/224-ui-changes-to-player-selection-screen)

File: Assets/Script/UI/MainMenu/ControllerSelection.cs [hunk 1]
```diff
@@ -97,6 +106,7 @@ private void GoToNextScreenIfNeeded()
             m_elapsedTimeToCancel -= Time.deltaTime;
             if (m_elapsedTimeToCancel <= 0)
             {
+                m_canPlayersJoin = false;
```
Comment (mikeslgoh) [inline Assets/Script/UI/MainMenu/ControllerSelection.cs]: what is this variable used for?

File: Assets/Script/UI/MainMenu/ControllerSelectionPanel.cs [hunk 2]
```diff
@@ -63,10 +69,10 @@ public class ControllerSelectionPanel : MonoBehaviour
     /// <param name="controllerName">Name of controller</param>
     /// <param name="loader">player data loader</param>
     /// <param name="player">player number</param>
-    public void InitializeController(Player player, string controllerName, PlayerDataLoader loader, InputDevice ip)
+    public void InitializeController(Player player, string controllerName, PlayerDataLoader loader, InputDevice ip, ref bool[] takenCars)
     {
         m_inputDevice = ip;
-        SetupPanel(controllerName, loader, player);
+        SetupPanel(controllerName, loader, player,ref takenCars);
```
Comment (mikeslgoh) [inline Assets/Script/UI/MainMenu/ControllerSelectionPanel.cs]: what does this 'ref' do?

File: Assets/Script/UI/MainMenu/CarSelectionReadyBar.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+using UnityEngine.UI;
+using TMPro;
+using InControl;
+
+[RequireComponent(typeof(Image))]
+public class CarSelectionReadyBar : MonoBehaviour
+{
+    private Color m_readyColor = AngryGooseGames.Utilities.ColorFrom255(255, 159, 67);
+    private Color m_choosingColor = AngryGooseGames.Utilities.ColorFrom255(123, 123, 123);
+    private const string m_pressToConfirm = "Press {0} to Confirm";
```
Comment (mikeslgoh) [inline Assets/Script/UI/MainMenu/CarSelectionReadyBar.cs]: Was wondering whether we could do this in C#. Might used this to refactor the Back Button message

File: Assets/Script/Data/PlayerDataLoader.cs [hunk 4]
```diff
@@ -72,7 +72,7 @@ public struct CarObj {
         public CarObj(Sprite sprite)
         {
             this.m_Sprite = sprite;
-            this.m_Color = Utilities.DominantColor(sprite.texture);           
+            this.m_Color = Utilities.AverageColor(sprite.texture);
```
Comment (mikeslgoh) [inline Assets/Script/Data/PlayerDataLoader.cs]: what is the difference between AverageColor and DominantColor?

File: Assets/Script/UI/MainMenu/ControllerSelectionPanel.cs (hunk 2 above)
Comment (m4rtint) [reply Assets/Script/UI/MainMenu/ControllerSelectionPanel.cs]: The array is referenced, not copied

File: Assets/Script/Data/PlayerDataLoader.cs (hunk 4 above)
Comment (m4rtint) [reply Assets/Script/Data/PlayerDataLoader.cs]: average is mean
Dominant is mode

In statistics average terms.

Comment (m4rtint) [reply Assets/Script/Data/PlayerDataLoader.cs]: Average matches the figma mocks more

File: Assets/Script/UI/MainMenu/ControllerSelection.cs (hunk 1 above)
Comment (m4rtint) [reply Assets/Script/UI/MainMenu/ControllerSelection.cs]: Stops players joining in the middle of animation, transitioning between screens

File: Assets/Script/UI/MainMenu/ControllerSelectionPanel.cs (hunk 2 above)
Comment (mikeslgoh) [reply Assets/Script/UI/MainMenu/ControllerSelectionPanel.cs]: Ah C related things

Comment (m4rtint) [reply Assets/Script/UI/MainMenu/ControllerSelectionPanel.cs]: It's super important for c++, I don't use it enough to Be an expert.

File: Assets/Script/UI/MainMenu/CarSelectionReadyBar.cs (hunk 3 above)
Comment (Gursaanj) [reply Assets/Script/UI/MainMenu/CarSelectionReadyBar.cs]: ooh did not know this was a thing!!. we could add to our list of things to refactor for later or I could do it right after I finish the SoundEffect PR

**Comment by mikeslgoh** [comment]
I just pushed the PNGs so I think if possible @m4rtint , try and make the changes for the instructions in this PR

**Comment by m4rtint** [comment]
I have some change sinhave yet to push. Will do that in an hour

**Comment by m4rtint** [comment]
I tested it out, everything is working well with controller as well

**Pull request merged**